src/js/client/editprefset-viewport.js

```diff
@@ -0,0 +1,1001 @@
+    });
+};
+
+/**
+ * Event Listener for when a button is clicked to delete
+ * a context. Fetches the context to delete from the elements
+ * data-contextid attribute.
+ *
+ * This listener doesn't actually delete the context, but
+ * opens the confirm delete context dialog.
+ *
+ * @param {gpii.devpmt.editPrefs} that - Main prefs editor component
+ * @param {Event} event - Browser event object
+ */
+gpii.devpmt.onDeleteContext = function (that, event) {
+    var contextId = event.currentTarget.dataset.contextid;
+    that.applier.change("activeModalDialog.contextId", contextId);
+    that.events.openConfirmDeleteContextDialog.fire();
+};
+
+/**
+ * Event Listener for when a button is clicked to edit
+ * a context. Fetches the context to delete from the elements
+ * data-contextid attribute.
+ *
+ * @param {gpii.devpmt.editPrefs} that - Main prefs editor component
+ * @param {Event} event - Browser event object
+ */
+gpii.devpmt.onEditContext = function (that, event) {
+    var contextId = event.currentTarget.dataset.contextid;
+    that.applier.change("activeModalDialog", {
+        contextId: contextId,
+        originalContextId: contextId,
+        contextName: that.model.flatPrefs.contexts[contextId].name
+    });
+    that.events.openEditContextDialog.fire();
+};
+
+gpii.devpmt.editPspShow = function (devpmt, event) {
+    var context = event.currentTarget.dataset.context;
+    var term = event.currentTarget.dataset.term;
+    var product = event.currentTarget.dataset.product;
+    var checked = event.currentTarget.checked;
+    var segs = ["contexts", context, "metadata", "psp", "show"];
+    if (product) {
+        segs.push(product.replace(/\./g, "\\."), term.replace(/\./g, "\\."));
+    }
+    else {
+        segs.push(term.replace(/\./g, "\\."));
+    }
+    var path = ["flatPrefs"].concat(segs).join(".");
+    devpmt.applier.change(path, checked);
+};
+
+gpii.devpmt.editPspMemory = function (devpmt, event) {
+    var context = event.currentTarget.dataset.context;
+    var term = event.currentTarget.dataset.term;
+    var product = event.currentTarget.dataset.product;
+    var checked = event.currentTarget.checked;
+    var segs = ["contexts", context, "metadata", "psp", "memory"];
+    if (product) {
+        segs.push(product.replace(/\./g, "\\."), term.replace(/\./g, "\\."));
+    }
+    else {
+        segs.push(term.replace(/\./g, "\\."));
+    }
+    var path = ["flatPrefs"].concat(segs).join(".");
+    devpmt.applier.change(path, checked);
+};
+
+gpii.devpmt.editValueEvent = function (that, event) {
+    that.locate("valueDisplayCell").removeClass("pmt-value-editing");
+    $(event.currentTarget).addClass("pmt-value-editing");
+    var newCurrent = {
+        context: event.currentTarget.dataset.context,
+        term: event.currentTarget.dataset.term,
+        //value: event.currentTarget.dataset.value,
+        product: event.currentTarget.dataset.product
+    };
+
+    var newMetadata = {};
+    if (!newCurrent.product) {
```

Comment:
This `if` block can be improved: line 908-911 is a duplicate of line 918-921.

Reply:
Thanks!  I moved the common lines outside the block